### PR TITLE
fixes Octokit rest cannot update data to gist

### PR DESCRIPTION
**Pull request by dahoba**
## PR-Codex overview

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const { getTopTracks } = require('./spotify')
 const { GH_TOKEN: github_token, GIST_ID: gist_id } = process.env
 
 const octo = new Octokit({
-  auth: `token ${github_token}`,
+  auth: `${github_token}`,
```
Comment (izayl) [inline index.js:10]: seems no need to modify this `token` string, maybe u local env is not valid?

**Review by izayl — CHANGES_REQUESTED** [review]
❤️ hi @dahoba, really appreciate this PR! 

the `token` seems no need to remove, u can check on your local machine, make sure u have a github token with create gist access.

File: index.js (hunk 1 above)
Comment (dahoba) [reply index.js:10]: I got an error in updateTopTracks step.  So I'm dig into the debug log, as you can see snippets below.

```
...
 spotify-box ran into an issue for updating your gist:
2023-10-21T10:16:50.1750320Z HttpError: Not Found
2023-10-21T10:16:50.1780886Z RequestError [HttpError]: Not Found
2023-10-21T10:16:50.1781985Z     at /home/runner/work/spotify-box/spotify-box/node_modules/@octokit/request/dist-node/index.js:66:23
2023-10-21T10:16:50.1783045Z     at processTicksAndRejections (node:internal/process/task_queues:96:5)
2023-10-21T10:16:50.1784053Z     at async updateTopTracks (/home/runner/work/spotify-box/spotify-box/index.js:67:5)
2023-10-21T10:16:50.1785085Z     at async main (/home/runner/work/spotify-box/spotify-box/index.js:16:3)
2023-10-21T10:16:50.1785950Z     at async /home/runner/work/spotify-box/spotify-box/index.js:96:5 {
2023-10-21T10:16:50.1786536Z   status: 404,
2023-10-21T10:16:50.1786828Z   headers: {
2023-10-21T10:16:50.1787218Z     'access-control-allow-origin': '*',
2023-10-21T10:16:50.1789579Z     'access-control-expose-headers': 'ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset',
2023-10-21T10:16:50.1791727Z     connection: 'close',
2023-10-21T10:16:50.1792511Z     'content-encoding': 'gzip',
2023-10-21T10:16:50.1793061Z     'content-security-policy': "default-src 'none'",
2023-10-21T10:16:50.1793721Z     'content-type': 'application/json; charset=utf-8',
2023-10-21T10:16:50.1794328Z     date: 'Sat, 21 Oct 2023 10:16:50 GMT',
2023-10-21T10:16:50.1795124Z     'referrer-policy': 'origin-when-cross-origin, strict-origin-when-cross-origin',
2023-10-21T10:16:50.1795857Z     server: 'GitHub.com',
2023-10-21T10:16:50.1796535Z     'strict-transport-security': 'max-age=31536000; includeSubdomains; preload',
2023-10-21T10:16:50.1797243Z     'transfer-encoding': 'chunked',
2023-10-21T10:16:50.1797818Z     vary: 'Accept-Encoding, Accept, X-Requested-With',
2023-10-21T10:16:50.1798399Z     'x-accepted-oauth-scopes': '',
2023-10-21T10:16:50.1798893Z     'x-content-type-options': 'nosniff',
2023-10-21T10:16:50.1799371Z     'x-frame-options': 'deny',
2023-10-21T10:16:50.1799887Z     'x-github-api-version-selected': '2022-11-28',
2023-10-21T10:16:50.1800688Z     'x-github-media-type': 'github.v3; format=json',
2023-10-21T10:16:50.1801384Z     'x-github-request-id': '9009:6C84:51D940:A9720B:6533A512',
2023-10-21T10:16:50.1801970Z     'x-oauth-scopes': 'repo',
2023-10-21T10:16:50.1802455Z     'x-ratelimit-limit': '100',
2023-10-21T10:16:50.1802906Z     'x-ratelimit-remaining': '98',
2023-10-21T10:16:50.1803386Z     'x-ratelimit-reset': '1697886405',
2023-10-21T10:16:50.1803896Z     'x-ratelimit-resource': 'gist_update',
2023-10-21T10:16:50.1804387Z     'x-ratelimit-used': '2',
2023-10-21T10:16:50.1804804Z     'x-xss-protection': '0'
2023-10-21T10:16:50.1805157Z   },
2023-10-21T10:16:50.1805415Z   request: {
2023-10-21T10:16:50.1805754Z     method: 'PATCH',
2023-10-21T10:16:50.1806541Z     url: 'https://api.github.com/gists/***',
2023-10-21T10:16:50.1807010Z     headers: {
2023-10-21T10:16:50.1807448Z       accept: 'application/vnd.github.v3+json',
2023-10-21T10:16:50.1808335Z       'user-agent': 'octokit-rest.js/18.0.12 octokit-core.js/3.2.4 Node.js/16.20.2 (linux; x64)',
2023-10-21T10:16:50.1809147Z       authorization: 'token [REDACTED]',
2023-10-21T10:16:50.1809751Z       'content-type': 'application/json; charset=utf-8'
2023-10-21T10:16:50.1810233Z     },
2023-10-21T10:16:50.1813846Z     body: '{"files":{"🎵 My Spotify Top Tracks":{"filename":"🎵 My Spotify Top Tracks","content":"1+1                                                AVA\\nFluxxwave (Slowed + Rev                   Clovis Reyes\\nBeauty and the Beast (f              Magic Philharmoni\\nThe Box                                    Roddy Ricch\\nCan You Feel the Love T              Magic Philharmoni\\nAll I Ever Wanted - fea              BALLISTIK BOYZ fr\\nglamorous x this girl -                       badkarma\\nThe Four Seasons (Conce               Baroque Festival\\nUnder the Sea (from \\"th              Magic Philharmoni\\nGrandfather Clock                   Nursery Rhymes 123"}}}',
2023-10-21T10:16:50.1816876Z     request: { hook: [Function: bound bound register] }
2023-10-21T10:16:50.1817369Z   },
2023-10-21T10:16:50.1817974Z   documentation_url: 'https://docs.github.com/rest/gists/gists#update-a-gist'
2023-10-21T10:16:50.1818606Z }
2023-10-21T10:16:50.1901907Z Post job cleanup.
2023-10-21T10:16:50.2973863Z [command]/usr/bin/git version
...
```
So I clone to my local machine, set all things up. If `token` keyword were in the value of the `auth:` the error still remain.
 
And I also check the octokit.js [API docs v18](https://octokit.github.io/rest.js/v18#authentication) or [API docs v20](https://octokit.github.io/rest.js/v20#authentication) appeared that `auth:` did not need `token` keyword in front of the github personal token.

Comment (izayl) [reply index.js:10]: @dahoba can u check if your .env file `GIST_ID` is the `31760769c9785f0c10eae4ad52388437` ?

this is from your [gist link](https://gist.github.com/dahoba/31760769c9785f0c10eae4ad52388437)

I have tested, it should work if gist_id is correct

Comment (dahoba) [reply index.js]: I'm sorry my mistaken :( It's working on my machine and via Github actions now.

**Review by izayl — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ async function updateTopTracks(json) {
     gist = await octo.gists.get({
       gist_id,
     })
+    // console.log('update success!')
```
[the remark this answers is not in the document]
Comment (dahoba) [reply index.js:39]: I've removed the line but PR already closed. Should I create another PR for this?

[the remark this answers is not in the document]
Comment (izayl) [reply index.js:39]: thanks, no need, already updated

**Pull request merged**